### ISSUE TITLE
Invalid URI Component in Path

### ISSUE THREAD

**Issue by james-portelli-cko** [tkoenig89/express-static-gzip#45]
Hi @tkoenig89 

I'm using this middleware to serve static compressed files using brotli compression, and came across an issue whereby if my server is probed with an invalid URI path then the middleware throws an error rather than it being caught and handled gracefully.

```  
app.use(
    '/',
    expressStaticGzip('dist/client', {
      enableBrotli: true,
      orderPreference: ['br'],
      index: false,
      serveStatic: {
        maxAge: 31536000000,
        immutable: true,
        setHeaders: (res, currPath) => {
          if (currPath.includes('/remoteEntry.js')) {
            res.setHeader('Cache-Control', 'public, max-age=0');
          }
        },
      },
    }),
  );
```

If I try to access the following URL on the server `http://localhost:5000/%c0` a 500 error is thrown with a URI Error: URI Malformed coming from the `decodeURIComponent` function being called [here](https://github.com/tkoenig89/express-static-gzip/blob/master/index.js#L32)

![image](https://user-images.githubusercontent.com/67912380/152537001-59d2c54d-97fa-4b0f-8823-b9074c0dc9ae.png)

Any idea on how I can resolve this?

Thanks
James

**Comment by tkoenig89** [comment]
Hey James,

I will have a look at this. In the meantime can you elaborate more on what you mean by 'gracefully'? So my first though was to return 400 in case someone sends a malformed uri, that leads to this kind of exception.

**Comment by james-portelli-cko** [comment]
Hi Tobias, 

Thanks for such a quick response!

Pretty much on the lines of what I was thinking, catching the error thrown by `decodeURIComponent` and returning a 400 response.

**Comment by tkoenig89** [comment]
I just added a testcase for this (UriError => 400). I'll publish a new version when I'm done.

**Comment by james-portelli-cko** [comment]
That's amazing thanks for such a quick turnaround Tobias :)

**Comment by tkoenig89** [comment]
You are welcome (and lucky, because i saw your message and had some spare time :))
Sorry did not want to close issue with my commit. Just tried out to link commit to issue. 
I just published version 2.1.2 with the fix. Please confirm, that it's working for you.

**Comment by james-portelli-cko** [comment]
Hey Tobias, just tested and its working for me :)

Just a note for a future release this warning is being thrown:

`express deprecated res.send(status, body): Use res.status(status).send(body) instead node_modules/express-static-gzip/index.js:36:17`

Thanks once again for all your help
